druid/src/widget/parse.rs

```diff
@@ -0,0 +1,65 @@
+use std::fmt::Display;
+use std::mem;
+use std::str::FromStr;
+
+use crate::kurbo::Size;
+use crate::{
+    BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx, UpdateCtx, Widget,
+};
+
+/// Converts a `Widget<String>` to a `Widget<Option<T>>`, mapping parse errors to None
```

Comment:
I would explicitly document that this works with the `FromStr` trait, and that `T` has to be a  type that implements `FromStr`.